### PR TITLE
HDDS-9490. Migrate tests in ozone-tools to JUnit5

### PR DESCRIPTION
**Pull request by raju-balpande**
## What changes were proposed in this pull request?
Migrate the following test classes to JUnit5

Please describe your PR in detail:
Migrate the following test classes to JUnit5. Covering following..
hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/audit/parser/TestAuditParser.java
hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/checknative/TestCheckNative.java
hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/conf/TestGetConfOptions.java
hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/debug/TestDBDefinitionFactory.java
hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/freon/TestContentGenerator.java
hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/freon/TestProgressBar.java
hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/freon/containergenerator/TestGeneratorDatanode.java
hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/genconf/TestGenerateOzoneRequiredConfigurations.java
hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneAddressClientCreation.java

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9490

## How was this patch tested?
Tested on local system and on by creating workflow action.

### CODE REVIEW

**Comment by adoroszlai** [comment]
Thanks @raju-balpande for the patch.  Please check unit test failure:

https://github.com/raju-balpande/apache_ozone/actions/runs/6726836943/job/18283984761#step:5:3122

**Comment by raju-balpande** [comment]
I knew this problem, In existing implementation the string was coming as "java.lang.ArrayIndexOutOfBoundsException: 5" whereas in my local testing it was coming as "java.lang.ArrayIndexOutOfBoundsException: Array index out of range: 5", so this is due to different implementation of ArrayIndexOutOfBoundsException.ArraygetMessage() in various versions.
I will modify it to be suitable in both the cases. So better I spit it in two checks, i.e.
If expected exception is ArrayIndexOutOfExcpetion and the getMessage conatains 5.

**Comment by raju-balpande** [comment]
The exception been raised is The Exception type is CommandLine$ExecutionException and the motive to check the cuase by ArrayIndexOutOfBoundsException on index 5. And hence changed the line:
`assertTrue(e.getMessage()
          .contains("java.lang.ArrayIndexOutOfBoundsException: 5"));`
to 
`      assertTrue(e.getCause() instanceof ArrayIndexOutOfBoundsException);
      assertTrue(e.getMessage().contains(": 5"));`
      
The reason for change is:
In some jdk version the message returned was `java.lang.ArrayIndexOutOfBoundsException: 5' and in some version it is `java.lang.ArrayIndexOutOfBoundsException: Array index out of range: 5`

**Comment by adoroszlai** [comment]
@Galsza @VarshaRaviCV please review

**Review by adoroszlai — APPROVED** [review]

File: hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/audit/parser/TestAuditParser.java [hunk 1]
```diff
@@ -19,12 +19,14 @@
...
-import org.junit.BeforeClass;
-import org.junit.Test;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/audit/parser/TestAuditParser.java:29]: Please change your IDE settings to keep static imports in a separate block.

**Comment by adoroszlai** [comment]
Thanks @raju-balpande for the patch.

When removing the last usage of JUnit4 in a submodule, please add `<allow.junit4>false</allow.junit4>` to the module's `pom.xml`, to prevent new JUnit4 tests being added in the future.  Please add it for `hadoop-ozone/tools` in one of your upcoming patches when you work on some other module.

**Pull request merged**